lib/utils/format.ts

```diff
@@ -0,0 +1,256 @@
+    });
+  }
+
+  if (options.where != null && isPlainObject(options.where)) {
+    out.where = mapWhereFieldNames(options.where, Model);
+  }
+
+  return out;
+}
+
+export function mapWhereFieldNames(where: Record<string | symbol, any>, Model: ModelStatic<Model>): object {
+  if (!where) {
+    return where;
+  }
+
+  const newAttributes: Record<string | symbol, any> = Object.create(null);
+  // note on 'as any[]': TypeScript < 4.4 does not support using Symbol for keys.
+  // Cast can be removed in sept. 2022 when we drop support for < 4.4
+  for (const attributeNameOrOperator of getComplexKeys(where) as any[]) {
+    const rawAttribute: ModelAttributeColumnOptions | undefined = Model.rawAttributes[attributeNameOrOperator];
+
+    // note on 'any': TypeScript < 4.4 does not support using Symbol for keys.
+    // Cast can changed back to 'symbol | string' in sept. 2022 when we drop support for < 4.4
+    const columnNameOrOperator: any = rawAttribute?.field ?? attributeNameOrOperator;
+
+    if (
+      isPlainObject(where[attributeNameOrOperator])
+        && !(
+          rawAttribute
+          && (rawAttribute.type instanceof DataTypes.HSTORE
+            || rawAttribute.type instanceof DataTypes.JSON)
+        )
+    ) {
+      // Prevent renaming of HSTORE & JSON fields
+      newAttributes[columnNameOrOperator] = mapOptionFieldNames(
+        {
+          where: where[attributeNameOrOperator],
+        },
+        Model,
+      ).where;
+
+      continue;
+    }
+
+    if (Array.isArray(where[attributeNameOrOperator])) {
+      newAttributes[attributeNameOrOperator] = [...where[attributeNameOrOperator]];
+
+      for (const [index, wherePart] of where[attributeNameOrOperator].entries()) {
+        if (isPlainObject(wherePart)) {
+          newAttributes[columnNameOrOperator][index] = mapWhereFieldNames(wherePart, Model);
+        }
+      }
+
+      continue;
+    }
+
+    newAttributes[columnNameOrOperator] = where[attributeNameOrOperator];
+  }
+
+  return newAttributes;
+}
+
+/**
+ * getComplexKeys
+ *
+ * @param  {object} obj
+ * @returns {Array<string|symbol>} All keys including operators
+ * @private
+ */
+export function getComplexKeys(obj: object): Array<string | symbol> {
+  return [
+    ...getOperators(obj),
+    ...Object.keys(obj),
+  ];
+}
+
+/**
+ * getOperators
+ *
+ * @param  {object} obj
+ * @returns {Array<symbol|string>} All operators properties of obj
+ * @private
+ */
+export function getOperators(obj: object): symbol[] {
+  return Object.getOwnPropertySymbols(obj).filter(s => operatorsSet.has(s));
+}
+
+export function combineTableNames(tableName1: string, tableName2: string): string {
+  return tableName1.toLowerCase() < tableName2.toLowerCase()
+    ? tableName1 + tableName2
+    : tableName2 + tableName1;
+}
+
+/**
+ * Used to map field names in values
+ *
+ * @param dataValues
+ * @param attributeNames
+ * @param ModelClass
+ */
+export function mapValueFieldNames( // TODO: rename to mapAttributesToColumNames?
```

Comment:
Is this something to be addressed in this PR?

Reply:
Yes, I put the TODO because I wanted external input :)

It's related to https://github.com/sequelize/meetings/issues/17
Or I can just not include it in this PR and list changes in the above issue

Reply:
I think this would fit better in that issue then. Or add the RFC issue to the TODO

Reply:
Done :)